test/jdk/java/lang/Thread/crac/JoinOrSleepOnCRPauseTest.java

```diff
@@ -0,0 +1,194 @@
+// Copyright 2019-2021 Azul Systems, Inc.  All Rights Reserved.
+// DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+//
+// This code is free software; you can redistribute it and/or modify it under
+// the terms of the GNU General Public License version 2 only, as published by
+// the Free Software Foundation.
+//
+// This code is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+// A PARTICULAR PURPOSE.  See the GNU General Public License version 2 for more
+// details (a copy is included in the LICENSE file that accompanied this code).
+//
+// You should have received a copy of the GNU General Public License version 2
+// along with this work; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+//
+// Please contact Azul Systems, 385 Moffett Park Drive, Suite 115, Sunnyvale,
+// CA 94089 USA or visit www.azul.com if you need additional information or
+// have any questions.
+
+/*
+ * @test JoinOrSleepOnCRPauseTest.java
+ * @requires (os.family == "linux")
+ * @library /test/lib
+ * @summary check if Thread.join(timeout) and Thread.sleep(timeout)
+ *          will be completed on restore immediately
+ *          if their end time fell on the CRaC pause period
+ *          (i.e. between the checkpoint and restore)
+ * @run main/othervm JoinOrSleepOnCRPauseTest
+ */
+
+
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+
+import java.util.concurrent.CountDownLatch;
+
+
+public class JoinOrSleepOnCRPauseTest {
+
+    private static enum TestType {join, sleep};
+    private final TestType testType;
+
+    private final boolean nsAccuracy;
+
+    private final static long EPS_NS = Long.parseLong(
+        System.getProperty("cractests.time.eps", "100000000")); // default: 0.1s
```

Comment:
Please use a name reflecting the current test package(path) and name `test.jdk.java.lang.Thread.crac.JoinOrSleepOnCRPauseTest.eps`

Reply:
ok

Reply:
p.s. (after some thoughts): if several similar test will be written, then it might be better to have some common option, in order not to pass all of them in jtreg, one per test..

Reply:
done